### PR TITLE
refactor: article list native

### PR DESCRIPTION
**Pull request by tuncaulubilge**
Further performance improvements on article list

- WindowSize on the flatlist has been reduced to 5 (default value is 21, which eagerly renders -10 +10 screens for smooth scrolling. We don't need such a high look-ahead value, so it has been reduced for better performance)
- RenderItem and onPress functions were being created for item on each render. Moved them to the component itself.

### CODE REVIEW

File: packages/article-list/src/article-list-item.js [hunk 1]
```diff
@@ -21,13 +26,20 @@ class ArticleListItem extends Component {
     );
   }
 
+  onItemPress(event) {
+    const {
+      article: { id, url },
+      onPress
+    } = this.props;
```
Comment (nathggns) [inline packages/article-list/src/article-list-item.js]: I really dislike that we have a lint rule that makes us do this.

File: packages/article-list/src/article-list.js [hunk 2]
```diff
@@ -78,6 +79,38 @@ class ArticleList extends Component {
...
+      imageRatio,
+      onArticlePress,
+      pageSize,
+      showImages
+    } = this.props;
+    const { width } = this.state;
+
+    return (
+      <ErrorView>
+        {({ hasError }) =>
```
Comment (nathggns) [inline packages/article-list/src/article-list.js]: This function still gets created every time. Is this a concern?

File: packages/article-list/src/article-list-item.js (hunk 1 above)
Comment (tuncaulubilge) [reply packages/article-list/src/article-list-item.js]: I do like destructing but being forced to do it everytime is a bit annoying for sure. Latest version of airbnb eslint had this forced in, but I think it's worth a discussion

File: packages/article-list/src/article-list.js (hunk 2 above)
Comment (tuncaulubilge) [reply packages/article-list/src/article-list.js]: :+1: I'll fix this

**Pull request merged**